lib/active_merchant/billing/gateways/xpay.rb

```diff
@@ -81,6 +82,19 @@ def scrub(transcript)
 
       private
 
+      def validation(options = {})
+        post = {}
+        add_3ds_validation_params(post, options)
+        commit(:validation, post, options)
+      end
+
+      def complete_transaction(action, amount, payment_method, options = {})
+        MultiResponse.run do |r|
+          r.process { validation(options) }
```

Comment:
💬 Comment:

Unless you know that server is always going to answer with a 200 on this end-point maybe a check here before calling the final step is a good idea.

Reply:
When the transaction is not successfully the next process is not executed because the following unless statement: https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/response.rb#L62

Then the response retrieve a fail response in the validation part.